src/emulator/io/include/io/file_wrapper.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <string>
+
+/// <summary>
```

Comment:
Remind you to not use c# doc type, but rather uses Doxygen doc type. Google it.

Reply:
Oh good to know, I'll go ahead and install it!!

Reply:
Sorry for not being clear, it's also a kind of document style. Here are [example](http://www.doxygen.nl/manual/docblocks.html)

Normally, we would do like this.

For functions, classes:

```cpp
/**
 * \brief Brief description about your documenting object.
 *
 * Detail description here.
 * \param name_param     Parameter usage description for function documentation
 * \param another_param Another parameter for example.
 *
 * \returns Documenting return value for function.
*/
````

For variables, enum documentation, we usually do it like this
```cpp
int var;   ///< This is a variable description

enum class example_enum {
   example_enum_value ///< Enum value description again
}
```

Reply:
I also was not clear, I meant that I'll go and install my tools to help with the correct formatting, did not mean to install documentation! But good to know, I'll format my stuff to follow the standard.